src/layers.rs

```diff
@@ -42,42 +43,145 @@ impl LayerTile {
     }
 }
 
-#[derive(Debug, PartialEq, Clone)]
+#[derive(Clone, PartialEq, Debug)]
+pub enum LayerType {
+    TileLayer(TileLayer),
+    ObjectLayer(ObjectLayer),
+    ImageLayer(ImageLayer),
+    // TODO: Support group layers
+}
+
+impl LayerType {
```

Comment:
I'm not entirely convinced we need this. It seems to me that it encourages use of `if` and `unwrap`, where I would expect in most cases it's best to do a `match`.

Reply:
Added these because of [the examples](https://github.com/mapeditor/rs-tiled/pull/131#discussion_r792079569), where I thought that matching turned out way too verbose. I can revert the changes if you want.

Reply:
We can leave that for now. It does appear useful there, but that is testing code, not example.